vault/activity_log.go

```diff
@@ -2350,11 +2366,16 @@ func (a *ActivityLog) segmentToPrecomputedQuery(ctx context.Context, segmentTime
 					{Name: "namespace", Value: a.namespaceToLabel(ctx, nsID)},
 				},
 			)
+			summedMetricsReporting[secretSyncActivityType] += entry.Counts.countByType(secretSyncActivityType)
 		}
 	}
 
-	// convert the maps to the proper format and write them as precomputed queries
-	return a.writePrecomputedQuery(ctx, segmentTime, opts)
+	for clientType, count := range summedMetricsMonthly {
+		a.metrics.SetGauge([]string{"identity", strings.ReplaceAll(clientType, "-", ""), "active", "monthly"}, float32(count))
+	}
+	for clientType, count := range summedMetricsReporting {
+		a.metrics.SetGauge([]string{"identity", strings.ReplaceAll(clientType, "-", ""), "active", "reporting_period"}, float32(count))
```

Comment:
Did you consider substituting `_` for rather than removing hyphens? I think our metrics often (though probably not consistentlty) use underscores for multi-word terms -- for example `reporting_period` in the same metric name here. I have a very gentle preference towards `identity.non_entity.active.reporting_period` rather than `identity.nonentity.active.reporting_period` on consistency grounds. But not a big deal if you disagree.

Reply:
I also prefer underscores. We already have the metric `nonentity` and so I was trying to make secret syncs follow the same pattern, but there's no reason why it needs to. I'll change it.